### PR TITLE
Resolves #195: Codify creation of standard derivatives into apps

### LINKED ISSUE #195: Codify creation of standard derivatives into apps
According to Slack discussions with @ianmilligan1 creation of standard derivates is still based on scripting AUT commands in Spark shell. We should probably codify into "apps" that can be called via `spark-submit`. This would also enable better e2e integration testing.

@ianmilligan1 why don't we start with the simplest such scripts that you're building - and I can start by mocking up what it would look like as an app.

### PR DESCRIPTION
**Pull request by TitusAn**
**Add RDD implementation and test cases for domain frequency extractor, domain graph extractor and plain text extractor. This provides references for data frame implementations of equivalent functionality.**

* * *

**GitHub issue(s)**: #195 

# Additional Notes:

Domain frequency extractor, domain graph extractor and plain text extractor are created, along with their test cases against warc/example.warc.gz.

org.rogach.scallop is used to parse command line arguments.

* Could this change or impact execution of existing code?

No changes to the existing code.

# Interested parties

Tag (@ mention) interested parties.

@lintool @ianmilligan1

### CODE REVIEW

**Comment by lintool** [comment]
What's the rationale for having both `apply` and `applyAndSave`? Is this some idiom I'm not familiar with?

**Comment by TitusAn** [comment]
applyAndSave is gone now. Should have taken that out.

File: src/main/scala/io/archivesunleashed/app/DomainFrequencyExtractor.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package io.archivesunleashed.app
```
Comment (lintool) [inline src/main/scala/io/archivesunleashed/app/DomainFrequencyExtractor.scala]: Add standard license header to this and all files.

**Comment by lintool** [comment]
Give @ianmilligan1 inline a sample command-line invocation to test - using `spark-submit`, right?

@greebie @ianmilligan1 test drive when you get a chance please.

**Comment by ianmilligan1** [comment]
Hey looks great! Aye, if you could give me the sample line I'd be happy to test that way.

**Comment by TitusAn** [comment]
./spark/bin/spark-submit --class io.archivesunleashed.app.DomainFrequencyExtractor ./aut/aut-0.16.1-SNAPSHOT-fatjar.jar --input example.warc.gz --output extractedDomainFrequency

This will run the extractor app on the input file, and output results in directory extractedDomainFrequency . Substitute DomainFrequencyExtractor in class option to DomainGraphExtractor or PlainTextExtractor to run different apps. Change path to spark-submit and aut executable as necessary.

**Comment by TitusAn** [comment]
Will start a brand new fork and recreate pull request to prevent this repo from being cluttered.

**Pull request closed without merging**